File: contracts/wasm/timestamp/src/timestamp.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use wasmlib::*;

use crate::*;

pub fn func_now(ctx: &ScFuncContext, f: &NowContext) {
    f.state.timestamp().set_value(ctx.timestamp());
}

pub fn view_get_timestamp(_ctx: &ScViewContext, f: &GetTimestampContext) {
    f.results.timestamp().set_value(f.state.timestamp().value());
}

File: contracts/wasm/timestamp/ts/timestamp/timestamp.ts
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import * as wasmlib from "wasmlib"
import * as sc from "./index";

export function funcNow(ctx: wasmlib.ScFuncContext, f: sc.NowContext): void {
}

export function viewGetTimestamp(ctx: wasmlib.ScViewContext, f: sc.GetTimestampContext): void {
    f.results.timestamp().setValue(ctx.timestamp());
}
